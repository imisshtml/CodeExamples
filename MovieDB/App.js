import React, { Component } from 'react';
import { Provider } from 'react-redux';
import store from './src/redux/store';
import MainNavigation from './src/components/MainNavigation';

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <MainNavigation />
      </Provider>
    );
  }
}