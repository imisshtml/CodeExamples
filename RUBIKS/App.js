import React from 'react';
import { StyleSheet, Text, View, Image, Animated, Easing } from 'react-native';
import { createStackNavigator, createBottomTabNavigator } from 'react-navigation';
import { Provider } from 'react-redux';
import Ionicons from 'react-native-vector-icons/Ionicons';
import {
  AdMobBanner, AdMobInterstitial, PublisherBanner, AdMobRewarded, Constants, Svg
} from 'expo';

import store from './store';
import PlayOnline from './src/PlayOnline';
import AuthScreen from './src/AuthScreen';
import HomeScreen from './src/HomeScreen';
import WelcomeScreen from './src/WelcomeScreen';
import LoginScreen from './src/LoginScreen';
import AccountScreen from './src/AccountScreen';
import StandingsScreen from './src/StandingsScreen';
import SocialScreen from './src/SocialScreen';
import Game3x3 from './src/GameScreen_3x3';
import ForfeitScreen from './src/ForfeitScreen';

const ADUNITID = `ca-app-pub-4846990712111779~3774124099`;
const BANNER_ID = `ca-app-pub-4846990712111779/8204323692`;

class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      lastRefresh: Date(Date.now()).toString(),
    }
    this.refreshScreen = this.refreshScreen.bind(this)
  }

  refreshScreen() {
    console.log('update');
    this.setState({ lastRefresh: Date(Date.now()).toString() })
  }

  _getCurrentRouteName(navState) {
    //this may not be needed
    if (navState.hasOwnProperty('index')) {
        this._getCurrentRouteName(navState.routes[navState.index])
    } else {
        console.log("Current Route Name:", navState.routeName)
        // can then save this to the state (I used redux)
        //store.dispatch(setCurrentRouteName(navState.routeName))
    }

}
  render() {
    const mainTabNavigator = createBottomTabNavigator(
      {
        GAME: HomeScreen,
        STANDINGS: StandingsScreen,
        SOCIAL: SocialScreen,
        ACCOUNT: AccountScreen
      },
      {
        lazy: false,
        navigationOptions: ({ navigation }) => ({
          tabBarIcon: ({ focused, horizontal, tintColor }) => {
            const { routeName } = navigation.state;
            if (routeName === 'GAME') {
              return <Image source={focused ? require('./assets/game_tab_icon_active.png') : require('./assets/game_tab_icon.png')} 
                      style={{height: 25, width: 25}}/>;
            } else if (routeName === 'STANDINGS') {
            return <Image source={focused ? require('./assets/standings_tab_icon_active.png') : require('./assets/standings_tab_icon.png')} 
                    style={{height: 25, width: 42}}/>;
            } else  if (routeName === 'SOCIAL') {
              return <Image source={focused ? require('./assets/social_tab_icon_active.png') : require('./assets/social_tab_icon.png')} 
                    style={{height: 25, width: 38}}/>;
            }else  if (routeName === 'ACCOUNT') {
              return <Image source={focused ? require('./assets/account_tab_icon_active.png') : require('./assets/account_tab_icon.png')} 
                      style={{height: 25, width: 25}}/>;
            }
          },
        }),
        tabBarOptions: {
          activeTintColor: '#6e73ff',
          inactiveTintColor: 'gray',
          labelStyle:{ fontWeight: 'bold'},
        },
      }
    );

    /*
    const gameTabNavigator = createBottomTabNavigator(
      {
        FORFEIT: ForfeitScreen,
        SCRAMBLE: Game3x3
      },
      {
        initialRouteName: 'SCRAMBLE',
        navigationOptions: ({ navigation }) => ({
          tabBarIcon: ({ focused, horizontal, tintColor }) => {
            const { routeName } = navigation.state;
            if (routeName === 'FORFEIT') {
              return <Image source={require('./assets/forfeit_tab_icon.png')} 
                      style={{height: 25, width: 25}}/>;
            } else if (routeName === 'SCRAMBLE') {
            return <Image source={require('./assets/mix_tab_icon.png')} 
                    style={{height: 25, width: 25}}/>;
            }
          },
          tabBarOnPress: ({ focused }) => {
                const { routeName } = navigation.state;
                console.log('test: '+this.state.lastRefresh);
                //this.refreshScreen();
          }
        }),
        tabBarOptions: {
          activeTintColor: 'gray',
          inactiveTintColor: 'gray',
          labelStyle:{ fontWeight: 'bold'},

        },
      }
    );
    */

    const RootStack = createStackNavigator(
      {
        Welcome: WelcomeScreen,
        Login: LoginScreen,
        Auth: AuthScreen,
        Home: mainTabNavigator,
        Play: Game3x3,
      },
      {
        initialRouteName: 'Home',
        navigationOptions:
        {
            headerStyle: { height: 0, backgroundColor: 'transparent', opacity: 1},
            headerLeft: null,
        },
        transitionConfig: () => ({
          containerStyle: {
          },
          transitionSpec: {
            duration: 0,
            easing: Easing.out(Easing.poly(4)),
            timing: Animated.timing,
          },
          screenInterpolator: sceneProps => {      
            const { position, scene } = sceneProps
            const thisSceneIndex = scene.index
            const opacity = position.interpolate({
              inputRange: [thisSceneIndex - 1, thisSceneIndex],
              outputRange: [0, 1],
            })
            return { opacity } 
          },
        }),
      }
    );


    return (
      <Provider store={store}>
          <RootStack 
            onNavigationStateChange={(prevState, newState) => {
              this._getCurrentRouteName(newState)
          }}
          />
      </Provider>
    );
  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default App;
