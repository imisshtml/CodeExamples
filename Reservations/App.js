import fetch from 'node-fetch';
import React, {Component} from 'react';
import { Root } from "native-base";
import ApolloClient from 'apollo-client';
import { ApolloProvider } from 'react-apollo';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { HttpLink } from 'apollo-link-http';

import MainNavigation from 'src/navigation/MainNavigation';
import { graphqlURL } from 'src/configs/config';

const cache = new InMemoryCache();
const link = new HttpLink({
  uri: graphqlURL,
  fetch: fetch,
});

const client = new ApolloClient({
  link,
  cache
});

export default class App extends Component {
  render() {
    return (
      <ApolloProvider client={client}>
        <Root>
          <MainNavigation />
        </Root>
      </ApolloProvider>
    );
  }
}