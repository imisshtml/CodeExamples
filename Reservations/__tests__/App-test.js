import 'react-native';
import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

jest.useFakeTimers();

jest.mock("NativeModules", () => ({
  UIManager: {
    RCTView: () => ({
      directEventTypes: {}
    })
  },
  KeyboardObserver: {},
  RNGestureHandlerModule: {
    attachGestureHandler: jest.fn(),
    createGestureHandler: jest.fn(),
    dropGestureHandler: jest.fn(),
    updateGestureHandler: jest.fn(),
    State: {},
    Directions: {}
  }
}))

test('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});