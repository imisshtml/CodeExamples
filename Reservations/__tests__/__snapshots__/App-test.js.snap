// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "paddingTop": 64,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "translateX": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            style={
              Object {
                "backgroundColor": "#fff",
                "flex": 1,
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4294963691,
                  4289519865,
                ]
              }
              endPoint={
                Object {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <RCTSafeAreaView
                emulateUnlessSupported={true}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <ActivityIndicator
                    animating={true}
                    color="#999999"
                    hidesWhenStopped={true}
                    size="large"
                  />
                </View>
              </RCTSafeAreaView>
            </BVLinearGradient>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            onLayout={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderBottomColor": "#A7A7AA",
                "borderBottomWidth": 0.5,
                "height": 64,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "flexDirection": "row",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    accessibilityTraits="header"
                    allowFontScaling={false}
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, .9)",
                        "fontSize": 17,
                        "fontWeight": "600",
                        "marginHorizontal": 16,
                        "textAlign": "left",
                      }
                    }
                  >
                    Reservations
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": 0,
            "maxHeight": 667,
            "minHeight": 56,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "marginHorizontal": -15,
              "marginTop": 0,
            }
          }
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
